pkgs/applications/networking/cluster/rke2/update-script.sh

```diff
@@ -1,49 +1,41 @@
 #!/usr/bin/env nix-shell
-#!nix-shell -i bash -p curl git gnugrep gnused yq-go nix-prefetch
+#!nix-shell -i bash -p curl git gnugrep gnused yq-go nix-prefetch go
 
-set -x -eu -o pipefail
+set -x -e -o pipefail
 
 CHANNEL_NAME="${1:?Must provide a release channel, like 'stable', as the only argument}"
 
 WORKDIR=$(cd $(dirname ${BASH_SOURCE[0]}) && pwd -P)
-
 mkdir --parents --verbose "${WORKDIR}/${CHANNEL_NAME}"
 
-LATEST_TAG_NAME=$(curl --silent --fail https://update.rke2.io/v1-release/channels | \
-    yq eval ".data[] | select(.id == \"${CHANNEL_NAME}\").latest" - | \
-    sort -rV | grep --extended-regexp "^v[0-9]+\.[0-9]+\.[0-9]+" | head -n1)
+LATEST_TAG_NAME=$(curl -sS --fail https://update.rke2.io/v1-release/channels | \
+    yq ".data[] | select(.id == \"${CHANNEL_NAME}\") | .latest")
 
 RKE2_VERSION=$(echo ${LATEST_TAG_NAME} | sed 's/^v//')
+RKE2_COMMIT=$(curl -sS --fail "https://api.github.com/repos/rancher/rke2/git/refs/tags/${LATEST_TAG_NAME}" | yq '.object.sha')
 
-RKE2_REPO_SHA256=$(nix-prefetch-url --quiet --unpack \
-        https://github.com/rancher/rke2/archive/refs/tags/${LATEST_TAG_NAME}.tar.gz)
-
-RKE2_COMMIT=$(curl --silent --fail ${GITHUB_TOKEN:+-u ":${GITHUB_TOKEN}"} \
-        https://api.github.com/repos/rancher/rke2/git/refs/tags | \
-    yq eval ".[] | select(.ref == \"refs/tags/${LATEST_TAG_NAME}\").object.sha" -)
-
-VERSIONS_SCRIPT=$(mktemp --suffix ".${RKE2_COMMIT:0:6}.sh")
-trap "rm --force ${VERSIONS_SCRIPT}" EXIT
-
-curl --silent --fail --output ${VERSIONS_SCRIPT} \
-        https://raw.githubusercontent.com/rancher/rke2/${RKE2_COMMIT}/scripts/version.sh
+PREFETCH_META=$(nix-prefetch-url --unpack --print-path "https://github.com/rancher/rke2/archive/refs/tags/${LATEST_TAG_NAME}.tar.gz")
+STORE_HASH="$(nix --extra-experimental-features nix-command hash to-sri --type sha256 ${PREFETCH_META%%$'\n'*})"
+STORE_PATH="${PREFETCH_META##*$'\n'}"
 
-set +eu
-DRONE_TAG=${LATEST_TAG_NAME} source ${VERSIONS_SCRIPT}
-set -eu
+cd ${STORE_PATH}
+GITHUB_ACTION_TAG=${LATEST_TAG_NAME}
+DRONE_COMMIT=${RKE2_COMMIT}
+source scripts/version.sh
+ETCD_BUILD=$(grep "images.DefaultEtcdImage" scripts/build-binary | sed 's/.*-\(build[0-9]*\)$/\1/')
+ETCD_VERSION="${ETCD_VERSION}-${ETCD_BUILD}"
+cd ${WORKDIR}
 
 KUBERNETES_CYCLES=$(echo ${KUBERNETES_VERSION} | grep -Eo "[0-9]+\.[0-9]+")
-KUBERNETES_EOL=$(curl --silent --fail \
-        https://endoflife.date/api/kubernetes/${KUBERNETES_CYCLES}.json | \
-    yq eval ".eol" -)
+KUBERNETES_EOL=$(curl -sS --fail https://endoflife.date/api/kubernetes/${KUBERNETES_CYCLES}.json | yq ".eol")
 
 FAKE_HASH="sha256-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA="
 
 cat << EOF > "${WORKDIR}/${CHANNEL_NAME}/versions.nix"
 {
   rke2Version = "${RKE2_VERSION}";
-  rke2RepoSha256 = "${RKE2_REPO_SHA256}";
```

Comment:
`builder.nix` does reference `rke2RepoSha256` in `fetchgit` when getting the `src` so I think removing this will break it no?
(same goes for all the version.nix files under each release dir.

Reply:
Yes, everything will fall apart without it. And especially OfBorg will be unhappy.

Reply:
I think this should be fine - all instances of `rke2RepoSha256` I could find have been replaced by `rke2TarballHash`, including the one in `fetchgit` (which has now been replaced by `fetchzip`). I'll mark it as resolved.